migrator/migrations/n_52_to_n_53_postgres_simple_access_scopes/migration.go

```diff
@@ -67,10 +132,129 @@ func move(gormDB *gorm.DB, postgresDB *pgxpool.Pool, legacyStore legacy.Store) e
 	return nil
 }
 
+func convertPermissionSetID(permissionSetID string) string {
+	identifierSuffix := strings.TrimPrefix(permissionSetID, permissionSetIDPrefix)
+	replacement, found := permissionsetIDMapping[identifierSuffix]
+	if found {
+		return replacement
+	}
+	_, err := uuid.FromString(identifierSuffix)
+	if err != nil {
+		generatedID := uuid.NewV4().String()
+		permissionsetIDMapping[identifierSuffix] = generatedID
+		return generatedID
+	}
+	return identifierSuffix
+}
+
+func movePermissionSets(gormDB *gorm.DB, postgresDB *pgxpool.Pool, legacyStore legacypermissionsets.Store) error {
+	ctx := sac.WithAllAccess(context.Background())
+	store := pgPermissionSetStore.New(postgresDB)
+
+	var permissionSets []*storage.PermissionSet
+	err := walkPermissionSets(ctx, legacyStore, func(obj *storage.PermissionSet) error {
+		permissionSetID := convertPermissionSetID(obj.GetId())
+		obj.Id = permissionSetID
+		permissionSets = append(permissionSets, obj)
+		if len(permissionSets) == batchSize {
+			if err := store.UpsertMany(ctx, permissionSets); err != nil {
+				log.WriteToStderrf("failed to persist permission_sets to store %v", err)
+				return err
+			}
+			permissionSets = permissionSets[:0]
+		}
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+	if len(permissionSets) > 0 {
+		if err = store.UpsertMany(ctx, permissionSets); err != nil {
+			log.WriteToStderrf("failed to persist permission_sets to store %v", err)
+			return err
+		}
+	}
+	return nil
+}
+
+func getRoleAccessScopeID(obj *storage.Role) (string, error) {
```

Comment:
nit: you're not required to do these changes, for me personally I'd like to be more specific about _which_ object we are dealing with. Applies to other functions as well (`getRolePermissionSetID`)
```suggestion
func getRoleAccessScopeID(role *storage.Role) (string, error) {
```